spec/factories/cart_factory.rb

```diff
@@ -1,7 +1,15 @@
 # frozen_string_literal: true
 
 FactoryBot.define do
+
   factory :cart do
     user
-  end
+
+    trait :without_user do
+      user{nil}
```

Comment:
```suggestion
      user {nil}
```

Espaço entre as declarações